fe/fe-core/src/main/java/org/apache/doris/planner/external/LoadPlanner.java

```diff
@@ -0,0 +1,202 @@
+        // Add an implicit container column "DORIS_DYNAMIC_COL" for dynamic columns
+        SlotDescriptor slotDesc = descTable.addSlotDescriptor(scanTupleDesc);
+        Column col = new Column(Column.DYNAMIC_COLUMN_NAME, Type.VARIANT, false, null, false, "",
+                "stream load auto dynamic column");
+        slotDesc.setIsMaterialized(true);
+        // Non-nullable slots will have 0 for the byte offset and -1 for the bit mask
+        slotDesc.setNullIndicatorBit(-1);
+        slotDesc.setNullIndicatorByte(0);
+        slotDesc.setColumn(col);
+        slotDesc.setIsNullable(false);
+        LOG.debug("plan scanTupleDesc{}", scanTupleDesc.toString());
+    }
+
+    public ScanNode scanNodeBuilder(TupleDescriptor scanTupleDesc, Database db, TUniqueId loadId) throws UserException {
+        FileLoadScanNode fileScanNode = new FileLoadScanNode(new PlanNodeId(0), scanTupleDesc);
+        // 1. create file group
+        DataDescription dataDescription = new DataDescription(table.getName(), taskInfo);
+        dataDescription.analyzeWithoutCheckPriv(db.getFullName());
+        BrokerFileGroup fileGroup = new BrokerFileGroup(dataDescription);
+        fileGroup.parse(db, dataDescription);
+        // 2. create dummy file status
+        TBrokerFileStatus fileStatus = new TBrokerFileStatus();
+        if (taskInfo.getFileType() == TFileType.FILE_LOCAL) {
+            fileStatus.setPath(taskInfo.getPath());
+            fileStatus.setIsDir(false);
+            fileStatus.setSize(taskInfo.getFileSize()); // must set to -1, means stream.
+        } else {
+            fileStatus.setPath("");
+            fileStatus.setIsDir(false);
+            fileStatus.setSize(-1); // must set to -1, means stream.
+        }
+        // The load id will pass to csv reader to find the stream load context from new load stream manager
+        fileScanNode.setLoadInfo(loadId, taskInfo.getTxnId(), table, BrokerDesc.createForStreamLoad(),
+                fileGroup, fileStatus, taskInfo.isStrictMode(), taskInfo.getFileType(), taskInfo.getHiddenColumns(),
+                taskInfo.isPartialUpdate());
+        ScanNode scanNode = fileScanNode;
+        scanNode.init(analyzer);
+        scanNode.finalize(analyzer);
+        scanNode.convertToVectorized();
+        return scanNode;
+    }
+
+    public ScanNode scanNodeBuilder(int nextNodeId,
+            TupleDescriptor scanTupleDesc,
+            long loadJobId,
+            long txnId,
+            BrokerDesc brokerDesc,
+            List<BrokerFileGroup> fileGroups,
+            boolean strictMode,
+            int loadParallelism,
+            UserIdentity userInfo) throws UserException {
+        ScanNode scanNode;
+        scanNode = new FileLoadScanNode(new PlanNodeId(nextNodeId), scanTupleDesc);
+        ((FileLoadScanNode) scanNode).setLoadInfo(loadJobId, txnId, table, brokerDesc, fileGroups,
+                fileStatusesList, filesAdded, strictMode, loadParallelism, userInfo);
```

Comment:
we should check fileStatusesList & filesAdded is already set, otherwise print a error log to tell other developer call preparePlan before plan